### PR TITLE
Sync native module functions are wrapping the result in an array, which it shouldn't

### PR DESCRIPTION
**Pull request by acoates-ms**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6253)

### CODE REVIEW

**Comment by NickGerleman** [comment]
I think we have a similar issue in some other places (e.g. https://github.com/microsoft/react-native-windows/issues/5951). Not sure if it's worth documenting in release notes, but I think this is technically breaking so I'm just going to label the PR in case we decide to document.

File: vnext/Microsoft.ReactNative.Cxx/NativeModules.h [hunk 1]
```diff
@@ -627,7 +627,7 @@ struct ModuleSyncMethodInfo<TResult (TModule::*)(TArgs...) noexcept>
       ArgTuple typedArgs{};
       ReadArgs(argReader, std::get<I>(typedArgs)...);
       TResult result = (module->*method)(std::get<I>(std::move(typedArgs))...);
-      WriteArgs(argWriter, result);
+      WriteValue(argWriter, result);
```
Comment (vmoroz) [inline vnext/Microsoft.ReactNative.Cxx/NativeModules.h:630]: >WriteValue [](start = 6, length = 10)

I would expect that we have tests for it and if it did not work, they must not pass. I wonder what scenario is currently broken.

**Comment by acoates-ms** [comment]
Need to add matching C# changes and update ReactModuleBuilderMock::CallSync to not unwrap the return value in the UTs.

**Review by vmoroz — APPROVED** [review]
:shipit:

**Pull request merged**